sys/net/gnrc/sock/udp/gnrc_sock_udp.c

```diff
@@ -24,14 +24,66 @@
 #include "net/gnrc/udp.h"
 #include "net/sock/udp.h"
 #include "net/udp.h"
-#include "random.h"
 
 #include "gnrc_sock_internal.h"
 
 #ifdef MODULE_GNRC_SOCK_CHECK_REUSE
 static sock_udp_t *_udp_socks = NULL;
 #endif
 
+static uint16_t _udp_dyn_port_next = 0;
+
+/* --------------------------------- helper --------------------------------- */
+
+/**
+ * @brief   Checks if a given UDP port is already used by another sock
+ */
+static bool _udp_dyn_port_used(uint16_t port)
+{
+#ifdef MODULE_GNRC_SOCK_CHECK_REUSE
+    for (sock_udp_t *ptr = _udp_socks; ptr != NULL;
+         ptr = (sock_udp_t *)ptr->reg.next) {
+        bool spec_addr = false;
+        for (unsigned i = 0; i < sizeof(ptr->local.addr); i++) {
+            const uint8_t *const p = (uint8_t *)&ptr->local.addr;
+            if (p[i] != 0) {
+                spec_addr = true;
+            }
+        }
+        if (spec_addr) {
+            continue;
+        }
+        if (ptr->local.port == port) {
+            /* port already in use by another sock */
+            return true;
+        }
+    }
+#endif /* MODULE_GNRC_SOCK_CHECK_REUSE */
+    return false;
+}
+
+/**
+ * @brief   returns a UDP port, and checks for reuse if required
+ */
+static uint16_t _udp_dyn_port(sock_udp_t *sock)
+{
+    uint16_t port;
+    uint16_t count = GNRC_UDP_DYN_PORTRANGE_NUM;
+    do {
+        port = GNRC_UDP_DYN_PORTRANGE_MIN +
+               (_udp_dyn_port_next + GNRC_UDP_DYN_PORTRANGE_OFF) % GNRC_UDP_DYN_PORTRANGE_NUM;
```

Comment:
Also, since both variable and these private functions are private and static, consider removing the `_udp` prefix for brevity (but keep the heading `_` to make their private nature clear).